src/cart/crt_swim.c

```diff
@@ -244,10 +313,26 @@ static int crt_swim_send_message(struct swim_context *ctx, swim_id_t to,
 	crt_rpc_t		*rpc_req;
 	crt_endpoint_t		 ep;
 	crt_opcode_t		 opc;
+	swim_id_t		 self_id = swim_self_get(ctx);
 	int			 opc_idx = 0;
 	int			 ctx_idx = csm->csm_crt_ctx_idx;
 	int			 rc;
 
+	if (nupds > 0 && upds[0].smu_state.sms_status == SWIM_MEMBER_INACTIVE)
+		opc_idx = 1;
+
+	opc = CRT_PROTO_OPC(CRT_OPC_SWIM_PROTO, CRT_OPC_SWIM_VERSION, opc_idx);
+
+	if (CRT_SWIM_SHOULD_FAIL(d_fa_swim_drop_rpc, self_id)) {
+		rc = -d_fa_swim_drop_rpc->fa_err_code;
+		D_ERROR("*** DROP outgoing opc %#x with %zu updates "
+			"%lu => %lu error: "DF_RC"\n", opc, nupds,
+			self_id, to, DP_RC(rc));
+		if (!rc)
+			D_FREE(upds);
```

Comment:
And are you selectively freeing something based on the contents of the config file?  Either it needs to be freed or it doesn't.

Reply:
This is required because of 'upds' should be free in completion callback or in case of error in upper layer. If we simulate success (rc != 0) we should simulate free like completion callback.

Reply:
That makes sense,  does that mean it also wants to be freed on other error paths, perhaps on L338 for example?

What would it mean if rc was set to zero here?  Would that mean the message is dropped, but anything else it gets replied to with the provided error code?

Reply:
As I mentioned before in case of error caller will free this. In case of error thare is no callback will be called therefore caller should free all data passed to this function.

Reply:
rc == 0 for this function means SWIM message was successfully sent and RPC completion callback will be called to free all data.